test/basic.html

```diff
@@ -18,130 +18,416 @@
+          assert.calledWith(playAnimationSpy, 'buffering');
         });
       });
     });
 
-    suite('transiting class', function() {
-      var progress;
+    suite('handling buffering animations when element is not yet attached', function() {
+      var container;
+      var playAnimationSpy;
 
       setup(function() {
-        progress = fixture('transitingProgress');
+        container = fixture('container');
+        progress = document.createElement('paper-linear-progress');
+        playAnimationSpy = sinon.spy(progress, 'playAnimation');
+        progress.buffering = true;
+
+        return flushPromise();
       });
 
-      test('progress bars', function() {
-        var stylesForPrimaryProgress = window.getComputedStyle(progress.$.primaryProgress);
-        var stylesForSecondaryProgress = window.getComputedStyle(progress.$.secondaryProgress);
-        var transitionProp = stylesForPrimaryProgress['transition-property'];
+      test('does not play the animation until the element is appended', function() {
+        assert.notCalled(playAnimationSpy);
+        container.appendChild(progress);
 
-        assert.isTrue(transitionProp === 'transform' || transitionProp === '-webkit-transform');
-        assert.equal(stylesForPrimaryProgress['transition-duration'], '0.08s');
+        return flushPromise().then(function() {
+          assert.called(playAnimationSpy);
+        });
+      });
 
-        transitionProp = stylesForSecondaryProgress['transition-property'];
+      test('does not play the animation if buffering is removed before attached', function() {
+        progress.buffering = false;
+        return flushPromise().then(function() {
+          container.appendChild(progress);
+          return flushPromise();
+        }).then(function() {
+          assert.notCalled(playAnimationSpy);
+        });
+      });
 
-        assert.isTrue(transitionProp === 'transform' || transitionProp === '-webkit-transform');
-        assert.equal(stylesForSecondaryProgress['transition-duration'], '0.08s');
+      test('always honors the latest toggle', function() {
+        progress.buffering = false;
+        return flushPromise().then(function() {
+          progress.buffering = true;
+          return flushPromise();
+        }).then(function() {
+          container.appendChild(progress);
+          return flushPromise();
+        }).then(function() {
+          assert.called(playAnimationSpy);
+        });
       });
     });
 
+    suite('closed', function() {
+      setup(function() {
+        progress = fixture('closed');
+      });
+
+      test('adds the closed class', function() {
+        assert.include(progress.className, 'closed');
+      });
+
+      test('removes the closed class when removed', function() {
+        progress.closed = false;
+        return flushPromise().then(function() {
+          assert.notInclude(progress.className, 'closed');
+        });
+      });
+    });
+
+    function flushPromise() {
```

Comment:
It would be helpful for these helpers to appear towards the top of the suite, or else in a separate `helpers.js`.